namer/marathon/src/main/scala/io/buoyant/namer/marathon/MarathonInitializer.scala

```diff
@@ -53,6 +62,15 @@ case class MarathonConfig(
 
   private[this] def getDst = dst.getOrElse(s"/$$/inet/$getHost/$getPort")
 
+  private[this] val secretKey = "DCOS_SERVICE_ACCOUNT_CREDENTIAL"
+  private[this] def getAuth =
+    sys.env.get(secretKey).flatMap { secret =>
+      Api.readJson[MarathonSecret](Buf.Utf8(secret)).toOption
+    }.collect {
```

Comment:
Is it appropriate to just ignore parse errors? I think it would be more user-friendly to fail to initialize when the environment variable is set but clearly invalid